### DISCUSSION TITLE
error with sample - user unauthorized but I can see the model in "My Models" and CONNECT

### DISCUSSION THREAD

**Discussion by s-o-w** [iTwin/itwinjs-core#3494] [General] [answered]
Having some issues getting my model to show.  I'm thinking probably related to some other scope I need to add?  the model is part of a project, for which my user account is valid.  But I get an unauthorized error.  Please advise - screenshots and env copied below.  I will recreate the app with the web-viewer CRA template once I know what's going on.

![screenshot](https://user-images.githubusercontent.com/20445178/162277247-28642fe2-3c37-40be-a814-10c0fad2d134.png)
![another](https://user-images.githubusercontent.com/20445178/162277362-493f5ebd-d1d5-43c6-a76d-fcafcc7a2c78.png)
![plant](https://user-images.githubusercontent.com/20445178/162278285-5efdff9d-1eee-49b6-af35-4f0616c6cf16.png)

```
IMJS_AUTH_CLIENT_CLIENT_ID = "spa-xxxxxxxxxxxxxxx"
IMJS_AUTH_CLIENT_REDIRECT_URI = "http://localhost:3000/signin-callback"
IMJS_AUTH_CLIENT_LOGOUT_URI = ""
IMJS_AUTH_CLIENT_SCOPES ="email openid profile organization itwinjs imodelaccess:read imodels:modify imodels:read library:modify library:read realitydata:modify realitydata:read savedviews:modify savedviews:read storage:modify storage:read projects:read projects:modify users:read"
IMJS_AUTH_AUTHORITY="https://ims.bentley.com"
IMJS_ITWIN_ID="2c574eae-8565-4fca-aef5-755f57cd32f2"
IMJS_IMODEL_ID="73886821-f5eb-431d-b0c5-e0bcab913abc"
SKIP_PREFLIGHT_CHECK=true
USE_FAST_SASS=true
USE_FULL_SOURCEMAPS=true
TRANSPILE_DEPS=false
DISABLE_TERSER=true
```

**Comment by ColinKerr** [comment, marked as answer]
If you are using iTwin.js 3.0 you only need `imodelaccess:read imodels:read realitydata:read` scopes to use the iTwin Viewer.

If you are not sure if you are using iTwin.js 3.0 just look in your package.json for @itwin packages that are 3.x.x versions.

**Reply by s-o-w** [reply]
I am using itwinjs > 3 and I have all of those scopes, but I still have a blocking access issue.
```
"dependencies": {
    "@bentley/icons-generic": "^1.0.34",
    "@bentley/react-scripts": "^4.0.7",
    "@itwin/appui-abstract": "^3.1.2",
    "@itwin/appui-layout-react": "^3.1.2",
    "@itwin/appui-react": "^3.1.2",
    "@itwin/browser-authorization": "^0.5.1",
    "@itwin/components-react": "^3.1.2",
    "@itwin/core-bentley": "^3.1.2",
    "@itwin/core-common": "^3.1.2",
    "@itwin/core-frontend": "^3.1.2",
    "@itwin/core-geometry": "^3.1.2",
    "@itwin/core-i18n": "^3.1.2",
    "@itwin/core-markup": "^3.1.2",
    "@itwin/core-orbitgt": "^3.1.2",
    "@itwin/core-quantity": "^3.1.2",
    "@itwin/core-react": "^3.1.2",
    "@itwin/core-telemetry": "^3.1.2",
    "@itwin/imodel-components-react": "^3.1.2",
    "@itwin/imodels-access-frontend": "^1.0.1",
    "@itwin/imodels-client-management": "^1.0.1",
    "@itwin/itwinui-react": "^1.35.0",
    "@itwin/presentation-common": "^3.1.2",
    "@itwin/presentation-frontend": "^3.1.2",
    "@itwin/reality-data-client": "^0.7.2",
    "@itwin/web-viewer-react": "^2.0.9",
    "@itwin/webgl-compatibility": "^3.1.2",
    "@types/history": "^4.7.11",
    "@types/node": "^14.18.12",
    "@types/react": "^17.0.44",
    "@types/react-dom": "^17.0.15",
    "history": "^4.10.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-redux": "^7.2.8",
    "typescript": "^4.3.5"
  },
```

**Reply by aruniverse** [reply]
Can you open the network tab, there should be a failing request that has a header for `x-correlation-id`. Could you share that with us?

**Reply by s-o-w** [reply]
![itworks](https://user-images.githubusercontent.com/20445178/162451552-60c31c67-1ee3-41a3-b0dd-4272892cac8a.png)

Well....after a reboot and a router reboot, it works now!  So, either I had some jacked up route, a network issue, or yall's presence scared it strait!  I'm going to call it good in either case, thank you so much!
